controllers/helper.go

```diff
@@ -311,6 +311,27 @@ func nodeSelectorTermsForPolicyList(policies []sriovnetworkv1.SriovNetworkNodePo
 		terms = append(terms, nodeSelector)
 	}
 
+	// sorting is needed to keep the daemon spec stable.
+	// the policies can come in a random order
+	sort.Slice(terms, func(i, j int) bool {
+		expressionsA := terms[i].MatchExpressions
+		expressionsB := terms[j].MatchExpressions
+		if len(expressionsA) != len(expressionsB) {
+			return len(expressionsA) < len(expressionsB)
+		}
+		for k := range expressionsA {
+			expressionA := expressionsA[k]
+			expressionB := expressionsB[k]
+			if expressionA.Key != expressionB.Key {
+				return expressionA.Key < expressionB.Key
+			}
+			if expressionA.Values[0] != expressionB.Values[0] {
+				return expressionA.Values[0] != expressionB.Values[0]
```

Comment:
```suggestion
				return expressionA.Values[0] < expressionB.Values[0]
```
Function returned bool value should indicate lower value and not that they are different. You can add the following test case to check it:
```
	terms := []corev1.NodeSelectorTerm{
		{MatchExpressions: []corev1.NodeSelectorRequirement{
			{Key: "env", Values: []string{"production"}},
		}},
		{MatchExpressions: []corev1.NodeSelectorRequirement{
			{Key: "env", Values: []string{"staging"}},
		}},
		{MatchExpressions: []corev1.NodeSelectorRequirement{
			{Key: "env", Values: []string{"dev"}},
		}},
	}
```

Output would be: `dev, staging, production`

Reply:
I also suggest to verify empty values, as it can lead to out of order index
```
            if expressionA.Key != expressionB.Key {
                return expressionA.Key < expressionB.Key
            }

            if len(expressionA.Values) == 0 {
                return true // Empty is considered less than non-empty
            }
            if len(expressionB.Values) == 0 {
                return false // Non-empty is considered greater than empty
            }
```

Reply:
Thank for you're review ! I'll add the test case for ordering base on value.
I think we expect to always have 1 value as it's create just above, that's why I was comparing only the first values.